lxc/config/remote.go

```diff
@@ -52,20 +51,95 @@ func (c *Config) ParseRemote(raw string) (remoteName string, resourceName string
 
 // GetInstanceServer returns a lxd.InstanceServer for the remote with the given name.
 func (c *Config) GetInstanceServer(name string) (lxd.InstanceServer, error) {
+	return c.GetInstanceServerWithAdditionalConnectionArgs(name, nil)
+}
+
+// GetInstanceServerWithAdditionalConnectionArgs returns a lxd.InstanceServer for the remote with the given name. Any
+// populated fields of the given connection arguments override the default connection arguments for the remote.
+func (c *Config) GetInstanceServerWithAdditionalConnectionArgs(name string, inArgs *lxd.ConnectionArgs) (lxd.InstanceServer, error) {
 	remote, err := c.getPrivateRemoteByName(name)
 	if err != nil {
 		return nil, err
 	}
 
 	// Get connection arguments
-	args, err := c.getConnectionArgs(name)
+	defaultArgs, err := c.getConnectionArgs(name)
 	if err != nil {
 		return nil, err
 	}
 
+	args := defaultArgs
+	if inArgs != nil {
+		args = mergeConnectionArgs(*defaultArgs, inArgs)
+	}
+
 	return c.connectRemote(*remote, args)
 }
 
+func mergeConnectionArgs(defaultArgs lxd.ConnectionArgs, customisableArgs *lxd.ConnectionArgs) *lxd.ConnectionArgs {
```

Comment:
why is customisableArgs a pointer?

Reply:
also we dont use compacted argument types.

Reply:
compacted argument types?

Reply:
```go
defaultArgs lxd.ConnectionArgs, customisableArgs *lxd.ConnectionArgs
```

rather than

```go
defaultArgs lxd.ConnectionArgs  *lxd.ConnectionArgs, customisableArgs *lxd.ConnectionArgs
```

and anyway in this case we cant use them because we should use:

```go
defaultArgs lxd.ConnectionArgs  *lxd.ConnectionArgs, customisableArgs lxd.ConnectionArgs
```

Reply:
@markylaing is there a more standard name for that?

Reply:
I'm sorry I'm still confused :confused: 

The first option you've mention is what is in the PR. The the second and third options don't look valid to me?

Reply:
updated it to make sense, i was rushing as need to work on other higher prio items

Reply:
its quite an odd function as its merging 2 structs,but one is pointer, suggesting its going to be modified, but it isn't, and the one that might be modified (the defaults) isn't a pointer, suggesting it won't be modified, but it is, internally anyway, and then it returns a modified copy of default args.

Quite for a code reviewer.

Reply:
Right I understand. Will fix in follow up after I've addressed https://github.com/canonical/lxd/pull/15311#discussion_r2044166402